### PR TITLE
chore: update deps

### PR DESCRIPTION
**Pull request by natapg**
This PR updates the following dependencies:

- Major updates:
  - `@fingerprintjs/fingerprintjs` from `^3.4.1` to `^4.0.0`
  - `@fingerprintjs/fingerprintjs-pro-react` from `^2.4.0` to `^2.5.0`
  - `simple-oauth2` from `^2.5.2` to `^5.0.0`
  - `swiper` from `^9.4.1` to `^10.0.4`
  - `typescript` from `^4.9.5` to `^5.1.6`
- Minor updates:
  - `@vercel/node` from `^2.15.2` to `^2.15.6`
  - `react-select` from `^5.3.2` to `^5.7.4`
  - `@graphql-eslint/eslint-plugin` from `3.10.7` to `3.20.1`
  - `@testing-library/jest-dom` from `5.16.5` to `5.17.0`
  - `@types/node` from `^20.3.1` to `^20.4.5`
  - `@types/react` from `18.2.15` to `18.2.17`
  - `@typescript-eslint/eslint-plugin` from `5.50.0` to `6.2.0`
  - `@typescript-eslint/parser` from `5.50.0` to `6.2.0`
  - `eslint` from `8.38.0` to `8.45.0`
  - `eslint-plugin-react` from `7.32.2` to `7.33.0`
  - `eslint-webpack-plugin` from `3.2.0` to `4.0.1`
  - `sass` from `1.63.6` to `1.64.1`

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ module.exports = {
       version: 'detect',
     },
   },
+  ...(!process.env.CI && { processor: '@graphql-eslint/graphql' }),
```
Comment (natapg) [inline .eslintrc.js:29]: we don't need to run this check on prod ci

File: api/_lib/oauth2.ts [hunk 2]
```diff
@@ -1,22 +1,24 @@
-import simpleOauthModule from 'simple-oauth2'
+import { AuthorizationCode } from 'simple-oauth2'
```
Comment (natapg) [inline api/_lib/oauth2.ts:1]: simpleOauthModule it not available on simple-oauth2 v5

File: src/components/Posts/Posts.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import Post, { PostProps } from '../Post/Post'
 import classNames from 'classnames'
 import TagList from '../TagList/TagList'
 import { Swiper, SwiperSlide } from 'swiper/react'
-import { Pagination } from 'swiper'
+import { Pagination } from 'swiper/modules'
```
Comment (natapg) [inline src/components/Posts/Posts.tsx:6]: https://swiperjs.com/migration-guide-v10#modules-imports

File: src/components/TagList/TagList.tsx [hunk 4]
```diff
@@ -49,7 +49,7 @@ export default function TagList({
             { [styles.vertical]: direction === 'vertical' }
           )}
         >
-          {tagLink || link ? (
+          {tagLink ?? link ? (
```
Comment (natapg) [inline src/components/TagList/TagList.tsx:52]: eslint complains about using Logical operators vs Nullish Coalescing

File: api/auth.ts [hunk 5]
```diff
@@ -5,14 +5,14 @@ import { create } from './_lib/oauth2'
 
 export const randomString = () => crypto.randomBytes(4).toString(`hex`)
 
-const auth = (req: VercelRequest, res: VercelResponse) => {
+const auth = async (req: VercelRequest, res: VercelResponse) => {
```
Comment (ctizen) [inline api/auth.ts:8]: Why async? 🤔 
Should there be `await` somewhere below?

File: src/hooks/useIntersectionObserver.ts [hunk 6]
```diff
@@ -20,7 +20,7 @@ function useIntersectionObserver(
     const node = elementRef?.current // DOM Ref
     const hasIOSupport = !!window.IntersectionObserver
 
-    if (!hasIOSupport || frozen || !node) return
+    if (!hasIOSupport ?? frozen ?? !node) return
```
Comment (ctizen) [inline src/hooks/useIntersectionObserver.ts:23]: This doesn't look like nullish coalescence. Maybe just add `!!` to `frozen` declaration instead?

**Comment by ctizen** [comment]
yarn.lock has shrunk nicely 🎉 . There's also things like [yarn-deduplicate](https://www.npmjs.com/package/yarn-deduplicate), you might want to take a look at it :) (not required to pass the review)

File: api/auth.ts (hunk 5 above)
Comment (natapg) [reply api/auth.ts:8]: at some point in the test I added an await, and then I deleted it ! it's not necessary at all, good catch

File: src/hooks/useIntersectionObserver.ts (hunk 6 above)
Comment (natapg) [reply src/hooks/useIntersectionObserver.ts]: yeah you are rigth , I've fixed it

**Review by ctizen — APPROVED** [review]

**Pull request merged**